scripts/checkpatch.sh

```diff
@@ -30,6 +30,29 @@ checkpatch_ignore+=",VOLATILE"            # Allow use of volatile keyword
 
 checkpatch_cmd+=" --ignore $checkpatch_ignore"
 
+# SPECIAL CASE: exclude tree.h
```

Comment:
We have the same sorts of checkpatch failures for the macros in `common/src/unifyfs_configurator.h`. Why have this in the checkpatch.sh script? The whole point of TEST_CHECKPATCH_SKIP_FILES is so you can set it in the commit message. You can just do `TEST_CHECKPATCH_SKIP_FILES=common/src/tree.h` in the commit message.

Reply:
I just tried that in the commit message, but it still flagged the errors.

Reply:
Did you do a rebase too?

Reply:
I did a `git commit --amend` to re-write the commit message.  I also just did a rebase, but still no luck.

Reply:
weird. I checked out your branch, removed the checkpatch change, and then did:
```
export TEST_CHECKPATCH_SKIP_FILES="common/src/tree.h"
git diff dev | ./scripts/checkpatch.sh
```
It worked for me.

Reply:
Yea, it works if you export it as a env var, but doesn't work if you add `TEST_CHECKPATCH_SKIP_FILES=common/src/tree.h` in the commit message.  That's why I had to add it to checkpatch.sh itself, to get it to pass Travis.

Reply:
I also just tried to amend the commit message, and that worked for me too. Wonder what is different in your environment?

Reply:
Is `TEST_CHECKPATCH_SKIP_FILES` still exported?  

I don't see anywhere in the code where checkpatch.sh would be looking through the commit message to set env variables.

Reply:
I recreated what travis does by the following commands:
```
> eval $(./scripts/git_log_test_env.sh)
> export TEST_CHECKPATCH_SKIP_FILES
> ./scripts/checkpatch.sh
> echo $TEST_CHECKPATCH_SKIP_FILES
common/src/tree.h
```

Reply:
It's the first command the greps for the TEST_* variables in the commit message and adds them into the environment.

Reply:
Ah ok, I wasn't aware Travis doing some magic beforehand when it called checkpatch.  That seems wrong, but that's an issue for another time.  I'll update the commit message with `TEST_CHECKPATCH_SKIP_FILES` set and just export it when I want to checkpatch manually.